src/test/java/com/microsoft/aad/msal4j/AadInstanceDiscoveryTest.java

```diff
@@ -135,4 +135,55 @@ public void aadInstanceDiscoveryTest_responseSetByDeveloper_invalidJson() throws
                 .aadInstanceDiscoveryResponse(instanceDiscoveryResponse)
                 .build();
     }
+
```

Comment:
Should there also be a test for the happy case, where region is detected, to validate URL is correctly formatted?

Reply:
I was working on something for the happy case, but couldn't think of anything useful to test since all of the endpoints are only reachable in a VM.

There's two main things that'd be good to test, but would require too much hardcoding to be useful tests:

- The IMDS call in `discoverRegion()`: The [IMDS endpoint](https://docs.microsoft.com/en-us/azure/virtual-machines/windows/instance-metadata-service?tabs=windows) is only accessible in an Azure VM, so we'd need to just mock the response to that endpoint ourselves. However, all we do is pull the region name from the body of the response, so all the test would do is confirm that `httpResponse.body()` returns the body that we mocked.
- The instance discovery call in `sendInstanceDiscoveryRequest()`: This endpoint is only available in certain regions and for certain clients, so we'd need to mock the response here too. And if we mock the response, then we're only testing whether the response we mocked can be parsed just like a response from the global endpoint, and I don't believe that the metadata format is different if it comes from a regional endpoint.

Reply:
Makes sense.